app/models/concerns/discovery_engine_nameable.rb

```diff
@@ -0,0 +1,31 @@
+# Enhances models with a `#name` method returning their fully qualified Google Cloud Platform
+# resource name (like a path).
+#
+# For example, for a `Control`, this would be:
+#   projects/{project}/locations/{location}/collections/{collection_id}/engines/
+#   {engine_id}/controls/{control_id}
+#
+# Requires the including class to implement `#discovery_engine_id`, and optionally `#parent` if the
+# parent resource is not the default collection.
+module DiscoveryEngineNameable
+  # The name (fully qualified path) of this Discovery Engine resource on GCP
+  def name
+    [parent_name, resource_path_fragment, discovery_engine_id].join("/")
```

Comment:
I can see why you've dried up these methods, but I'm, tripping up over the names. Is `discovery_engine_id` always the Engine id? I think in for Data stores it's the Data Store id? But then this concern is called `DiscoveryEngineNameable` but I think that's because of all of these particular Google products are namespaced with "DiscoveryEngine" even the Data Stores?